File: category/brute/2231.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	var strN string
	fmt.Scanf("%s\n", &strN)
	inputN, _ := strconv.Atoi(strN)

	offset := len(strN) * 9

	num, _ := strconv.Atoi(strN)
	var rst int
	for i := num - offset; i < num+offset; i++ {
		strNum := strconv.Itoa(i)

		var sum int = i
		for _, digit := range strNum {
			n, _ := strconv.Atoi(string(digit))

			sum += n
		}

		if sum == inputN {
			rst = i
			break
		}
	}
	fmt.Println(rst)
}

File: category/brute/2231.py

def main():
    strn = input()
    inputn = int(strn)

    offset = len(strn) * 9
    num = int(strn)
    rst = 0

    for i in range(max(1, num-offset), num):
        strnum = str(i)
        sum = i
        for digit in strnum:          
            sum += int(digit)
            
        
        if sum == inputn:
            rst = i
            break

    print(rst)
    

if __name__ == "__main__":
    main()